azure-pipelines.ppe.yml

```diff
@@ -0,0 +1,77 @@
+# Node.js with React
+# Build a Node.js project that uses React.
+# Add steps that analyze code, save build artifacts, deploy, and more:
+# https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
+
+trigger:
+  - develop
+
+pool:
+  vmImage: 'vs2017-win2016'
+
+steps:
+  # Azure Key Vault
+  # Download Azure Key Vault secrets
+  - task: AzureKeyVault@1
+    inputs:
+      azureSubscription: 'hazzat-api-ppe - Azure'
+      keyVaultName: 'hazzat-api-ppe'
+      secretsFilter: 'hazzatazure-ConnectionString'
+
+  - task: NodeTool@0
+    inputs:
+      versionSpec: '10.x'
+    displayName: 'Install Node.js'
+
+  - task: Npm@1
+    displayName: 'npm install'
+    inputs:
+      command: custom
+      workingDir: src
+      verbose: false
+      customCommand: 'install'
+
+  - task: Npm@1
+    displayName: 'npm build'
+    inputs:
+      command: custom
+      workingDir: src
+      verbose: false
+      customCommand: 'run build'
+
+  - task: Npm@1
+    displayName: 'npm test'
+    inputs:
+      command: custom
+      workingDir: src
+      verbose: false
+      customCommand: 'run test'
+    env:
+      CUSTOMCONNSTR_DBCONNECTIONSTRING: $(hazzatazure-ConnectionString)
+
+  - task: PowerShell@1
+    displayName: 'Generate Deployment Archive'
+    inputs:
+      scriptName: .build/createDeploymentArchive.ps1
+    env:
+      ARCHIVE_FILE_PATH: $(Build.ArtifactStagingDirectory)\$(Build.BuildId).zip
+      ENLISTMENT_ROOT: $(System.DefaultWorkingDirectory)
+
+  - task: CopyFiles@2
+    displayName: 'Copy File to: $(TargetFolder)'
+    inputs:
+      SourceFolder: '$(Build.ArtifactStagingDirectory)'
+      Contents: '$(Build.BuildId).zip'
+      TargetFolder: '$(Build.ArtifactStagingDirectory)\ArtifactsToBePublished'
+
+  - task: AzureWebApp@1
```

Comment:
I wanted a pipeline for build/test, and another pipeline for release.  This part should belong in the release pipeline instead.

Need to separate the build/test from the deployment.

Reply:
This means that the build pipeline would build and publish the built artifacts.
The release pipeline would download the artifacts and deploy them.  Not build again.

Reply:
I'm assuming that this should also take into account build configurations and deploy targets?

Reply:
Given test and release are the two pipelines with an artifact build, we can set up two pairs of variable groups for test/release: a variable group pair for the build pipeline, and a variable group pair for release. The PR build pipeline would presumably use the same variable group as test, but not continue to release (as it does currently). This will also allow test and release artifact builds to avoid naming collisions, as a variable can be set for the artifact name as well. 

We will need the following variable names for both variable group pairs on each file:

 ###### azure-pipelines.build.yaml
-    secretsFilter: $(SecretsFilter)
-    buildCommand: $(buildMode) 

 ###### azure-pipelines.release.yaml
- 

[link to docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml-task)

Reply:
I think we need a phone call to discuss.